src/main/java/com/mixpanel/android/viewcrawler/ViewCrawler.java

```diff
@@ -518,38 +534,33 @@ private void sendDeviceInfo() {
                         j.name(entry.getKey()).value(entry.getValue());
                     }
 
-                    final Map<String, Tweaks.TweakDescription> tweakDescs = mTweaks.getDescriptions();
+                    final Map<String, Tweaks.TweakValue> tweakDescs = mTweaks.getAllValues();
                     j.name("tweaks").beginArray();
-                    for (Map.Entry<String, Tweaks.TweakDescription> tweak:tweakDescs.entrySet()) {
-                        final Tweaks.TweakDescription desc = tweak.getValue();
+                    for (Map.Entry<String, Tweaks.TweakValue> tweak:tweakDescs.entrySet()) {
+                        final Tweaks.TweakValue desc = tweak.getValue();
                         final String tweakName = tweak.getKey();
                         j.beginObject();
                         j.name("name").value(tweakName);
-                        j.name("minimum").value(desc.minimum);
-                        j.name("maximum").value(desc.maximum);
+                        j.name("minimum").value((Number) null);
+                        j.name("maximum").value((Number) null);
                         switch (desc.type) {
-                            case Tweaks.UNKNOWN_TYPE:
-                                final Object currentValue = mTweaks.get(tweakName);
-                                j.name("type").value("unknown");
-                                j.name("value").value(null == currentValue ? null : currentValue.toString());
-                                break;
                             case Tweaks.BOOLEAN_TYPE:
                                 j.name("type").value("boolean");
-                                j.name("value").value(mTweaks.getBoolean(tweakName));
+                                j.name("value").value(desc.getBooleanValue());
                                 break;
                             case Tweaks.DOUBLE_TYPE:
                                 j.name("type").value("number");
                                 j.name("encoding").value("d");
-                                j.name("value").value(mTweaks.getDouble(tweakName));
+                                j.name("value").value(desc.getNumberValue().doubleValue());
                                 break;
                             case Tweaks.LONG_TYPE:
                                 j.name("type").value("number");
                                 j.name("encoding").value("l");
-                                j.name("value").value(mTweaks.getLong(tweakName));
+                                j.name("value").value(desc.getNumberValue().longValue());
                                 break;
                             case Tweaks.STRING_TYPE:
-                                j.name("type").value("string");
-                                j.name("value").value(mTweaks.getString(tweakName));
+                                j.name("type").value("unknown");
```

Comment:
Why is this guy `unknown`? I guess you are using it as a template and convert this `"value"` on the web UI. But, is web UI able to handle it? How can web UI tell wether it's a real string or something need to be converted?